lib/contentful/resource_builder.rb

```diff
@@ -182,8 +182,23 @@ def add_to_known_resources(res)
       @known_resources[res.type][res.id] = res if res.sys && res.id && res.type != 'Link'
     end
 
+    def localized_entry?(object, property_name, potential_objects)
+      object['sys']['type'] == 'Entry' &&
+        property_name == 'fields' &&
+        potential_objects.is_a?(::Hash) &&
+        potential_objects.any? { |_, p| Support.localized?(p) }
+    end
+
     def replace_children(res, object)
       object.each do |name, potential_objects|
+        if localized_entry?(object, name, potential_objects)
```

Comment:
Should the logic in this block be broken out into its own method (e.g. `detect_localized_objects`)? `replace_children` is starting to get a bit complicated with this addition.